### PR TITLE
Add `confluent schema-registry config delete` command

### PR DESCRIPTION
**Pull request by MuweiHe**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- Add `confluent schema-registry config delete` command

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

add a command to allow deleting top level or subject level compatibility config and reverting that to the default.

References
----------

https://github.com/confluentinc/schema-registry-sdk-go/blob/master/docs/DefaultApi.md#deletesubjectconfig

Test & Review
-------------

```
mhe@C02DV0KVMD6T cli % confluent schema-registry cluster update --compatibility full --compatibility-group application.version
Successfully updated top-level compatibility to "FULL".
mhe@C02DV0KVMD6T cli % confluent sr config delete                                                                             
Are you sure you want to delete the global compatibility level config and revert it to the default? (y/n): y
+---------------------+---------------------+
| Compatibility Level | FULL                |
| Compatibility Group | application.version |
+---------------------+---------------------+
mhe@C02DV0KVMD6T cli % confluent sr config describe
+---------------------+----------+
| Compatibility Level | BACKWARD |
+---------------------+----------+
```

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/cmd/schema-registry/command_config.go [hunk 1]
```diff
@@ -15,6 +20,19 @@ func (c *command) newConfigCommand(cfg *v1.Config) *cobra.Command {
 	}
 
 	cmd.AddCommand(c.newConfigDescribeCommand(cfg))
+	cmd.AddCommand(c.newConfigDeleteCommand(cfg))
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config.go:23]: Please alphabetize!

File: internal/cmd/schema-registry/command_config_delete.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+func (c *command) newConfigDeleteCommand(cfg *v1.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete top-level or subject-level schema configuration.",
+		Args:  cobra.NoArgs,
+		RunE:  c.configDelete,
+	}
+
+	example1 := examples.Example{
+		Text: `Delete the configuration of subject "payments".`,
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config_delete.go:25]:
```suggestion
		Text: `Delete the subject-level configuration of subject "payments".`,
```

File: internal/cmd/schema-registry/command_config_delete.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+	if cfg.IsCloudLogin() {
+		// Deprecated
+		pcmd.AddApiKeyFlag(cmd, c.AuthenticatedCLICommand)
+		cobra.CheckErr(cmd.Flags().MarkHidden("api-key"))
+
+		// Deprecated
+		pcmd.AddApiSecretFlag(cmd)
+		cobra.CheckErr(cmd.Flags().MarkHidden("api-secret"))
+	}
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config_delete.go:57]: No need to add deprecated flags!
```suggestion
```

File: internal/pkg/schema-registry/client.go [hunk 4]
```diff
@@ -47,6 +47,11 @@ func (c *Client) GetTopLevelConfig() (srsdk.Config, error) {
 	return res, err
 }
 
+func (c *Client) DeleteTopLevelConfig() (string, error) {
+	res, _, err := c.DefaultApi.DeleteTopLevelConfig(c.context)
```
Comment (brianstrauch) [inline internal/pkg/schema-registry/client.go:51]: So these APIs return a JSON string? Can we update the SDK to unmarshal this string into a struct?

File: internal/cmd/schema-registry/command_config_delete.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+	example1 := examples.Example{
+		Text: `Delete the subject-level configuration of subject "payments".`,
+		Code: "confluent schema-registry config delete --subject payments",
+	}
+	example2 := examples.Example{
+		Text: "Delete the top-level configuration.",
+		Code: "confluent schema-registry config delete",
+	}
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config_delete.go:31]: Nit: Let's swap these examples, so the example with the top-level configuration is on top

File: test/fixtures/output/schema-registry/config/delete-config.golden [hunk 6]
```diff
@@ -0,0 +1,3 @@
++---------------------+----------+
+| Compatibility Level | BACKWARD |
```
Comment (brianstrauch) [inline test/fixtures/output/schema-registry/config/delete-config.golden:2]: Shouldn't we print a message along the lines of "Schema Registry configuration successfully deleted" to match the style of our other commands?

File: internal/pkg/resource/resource.go [hunk 7]
```diff
@@ -29,6 +29,7 @@ const (
 	Pipeline              = "pipeline"
 	SchemaExporter        = "schema exporter"
 	SchemaRegistryCluster = "Schema Registry cluster"
+	SchemaRegistryConfig  = "Schema Registyr config"
```
Comment (brianstrauch) [inline internal/pkg/resource/resource.go:32]:
```suggestion
	SchemaRegistryConfiguration  = "Schema Registry configuration"
```

File: internal/cmd/schema-registry/command_config_delete.go [hunk 8]
```diff
@@ -81,6 +82,7 @@ func (c *command) configDelete(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	output.Printf("Deleted %s.\n", resource.SchemaRegistryConfig)
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config_delete.go:85]: We should match the style of other delete commands and say "Deleted Schema Registry configuration XXX". I think there's a global constant for this.

Comment (MuweiHe) [reply internal/cmd/schema-registry/command_config_delete.go:85]: yeah but it's easier to see the deleted configs as a table.

File: internal/pkg/resource/resource.go (hunk 7 above)
Comment (MuweiHe) [reply internal/pkg/resource/resource.go:32]: `config` is how the route is referenced in docs, and also used in descriptions. wdyt?

Comment (brianstrauch) [reply internal/pkg/resource/resource.go:32]: `config` is a common shorthand, but it doesn't hurt to use the full word and it's a bit more explicit / professional looking, IMO.

File: internal/cmd/schema-registry/command_config_delete.go (hunk 8 above)
Comment (brianstrauch) [reply internal/cmd/schema-registry/command_config_delete.go:85]: Aren't we only deleting one configuration here? Why would we need a table?

Comment (MuweiHe) [reply internal/cmd/schema-registry/command_config_delete.go:85]: configuration includes a --compatibility-group field too

Comment (MuweiHe) [reply internal/cmd/schema-registry/command_config_delete.go]:
```
mhe@C02DV0KVMD6T cli % confluent sr config delete --force                                                                         
Deleted Schema Registry configuration.
+---------------------+---------------------+
| Compatibility Level | BACKWARD            |
| Compatibility Group | application.version |
+---------------------+---------------------+
```

File: internal/pkg/schema-registry/client.go (hunk 4 above)
Comment (MuweiHe) [reply internal/pkg/schema-registry/client.go:51]: Their team don't want to make this breaking change of api 😕

File: test/test-server/schema_registry_handlers.go [hunk 9]
```diff
@@ -42,6 +42,10 @@ func handleSRUpdateTopLevelConfig(t *testing.T) http.HandlerFunc {
 			}
 			err := json.NewEncoder(w).Encode(res)
 			require.NoError(t, err)
+		case http.MethodDelete:
+			res := `{"compatibilityLevel":"BACKWARD"}`
```
Comment (brianstrauch) [inline test/test-server/schema_registry_handlers.go:46]: Can we serialize a struct here instead?

Comment (MuweiHe) [reply test/test-server/schema_registry_handlers.go:46]: this matches the behavior of the backend, returning a string. It'll be unmarshaled into a struct in the command code

Comment (brianstrauch) [reply test/test-server/schema_registry_handlers.go:46]: Oh right, I forgot that this API was broken. Write a comment?

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**